src/core/client/admin/components/UserStatus/SuspendModal.tsx

```diff
@@ -0,0 +1,195 @@
+import { Localized } from "fluent-react/compat";
+
+import { Format, withFormat } from "coral-framework/lib/i18n";
+import React, { FunctionComponent, useState } from "react";
+
+import NotAvailable from "coral-admin/components/NotAvailable";
+import {
+  Button,
+  Card,
+  CardCloseButton,
+  CheckBox,
+  Flex,
+  HorizontalGutter,
+  Modal,
+  RadioButton,
+  Typography,
+} from "coral-ui/components";
+import SuspendSuccessModal from "./SuspendSuccessModal";
+
+import styles from "./SuspendModal.css";
+
+interface Props {
+  username: string | null;
+  open: boolean;
+  onClose: () => void;
+  onSuccessClose: () => void;
+  onConfirm: (timeout: number, message: string) => void;
+  format: Format;
+  organizationName: string;
+  success: boolean;
+}
+
+const DURATIONS: Array<[string, string]> = [
+  ["3600", "1 hour"],
+  ["10800", "3 hours"],
+  ["86400", "24 hours"],
+  ["604800", "7 days"],
+];
+
+const DEFAULT_DURATION_INDEX = 1;
+
+const SuspendModal: FunctionComponent<Props> = ({
+  open,
+  onClose,
+  onConfirm,
+  username,
+  format,
+  success,
+  onSuccessClose,
+  organizationName,
+}) => {
+  function getMessageWithDuration(index: number): string {
+    return format("community-suspendModal-emailTemplate", {
+      username,
+      organizationName,
+      duration: DURATIONS[index][1],
+    });
+  }
+  const [showMessage, setShowMessage] = useState(false);
+  const [durationIndex, setDurationIndex] = useState(DEFAULT_DURATION_INDEX);
+  const [messageDirty, setMessageDirty] = useState(false);
+  const [emailMessage, setEmailMessage] = useState(
+    getMessageWithDuration(DEFAULT_DURATION_INDEX)
+  );
+
+  function selectDuration(index: number): void {
+    setDurationIndex(index);
+    if (!messageDirty) {
```

Comment:
need to keep track of whether message has been edited, because if it has we shouldn't update its contents if duration is changed